### ISSUE TITLE
Issue with >1.33.1 on Windows

### ISSUE THREAD

**Issue by leidegre** [c-ares/c-ares#908]
We encountered an issue with c-ares on Windows specifically, there is some change after version 1.33.1 that causes asserts to fail in gRPC on Windows but not Linux.

Please see this issue for more information, https://github.com/grpc/grpc/issues/37969 (gRPC)

I'm posting this with the hopes that we can zero in on what the change is and if it can be fixed in c-ares?

**Comment by bradh352** [comment]
I replied to that ticket, this appears to be a bug in grpc with an inverted sanity check.  We started using `struct sockaddr_storage` as the argument passed to `recvfrom()`, which is the recommended data type.  Previously we created a private union and passed that.  Apparently on Windows there is a size difference (structure is larger) and grpc doesn't like this due to a logic inversion flaw.  Passing a larger buffer to be filled in should never result in a failure.

I'm going to close this as there is clearly no bug in c-ares.
